diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index 7b7cee89f6..0f814dcbbb 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -179,8 +179,8 @@ Image.%: Image Image.gz
 	$(Q)$(MAKE) $(build)=$(boot) $(bootvars-y) $(boot)/$@
 
 #BPI
-#install: KBUILD_IMAGE := $(boot)/Image
-install: KBUILD_IMAGE := $(boot)/Image.itb
+install: KBUILD_IMAGE := $(boot)/Image
+#install: KBUILD_IMAGE := $(boot)/Image.itb
 zinstall: KBUILD_IMAGE := $(boot)/Image.gz
 install zinstall:
 	$(call cmd,install)
diff --git a/drivers/misc/icm42607.c b/drivers/misc/icm42607.c
index 8ac6532075..cd46edcf5c 100644
--- a/drivers/misc/icm42607.c
+++ b/drivers/misc/icm42607.c
@@ -553,7 +553,7 @@ struct attribute_group icm42607_attrs = {
 	.attrs = icm42607_sysfs_attrs,
 };
 
-int icm42607_probe(struct i2c_client *client, const struct i2c_device_id *devid)
+int icm42607_probe(struct i2c_client *client)
 {
 	struct icm42607_device *mpu;
 	int err;
diff --git a/drivers/spi/spi-dw-espi.c b/drivers/spi/spi-dw-espi.c
index 3bb27be374..87dda10266 100644
--- a/drivers/spi/spi-dw-espi.c
+++ b/drivers/spi/spi-dw-espi.c
@@ -1180,7 +1180,7 @@ int dw_spi_ext_add_host(struct device *dev, struct dw_spi *dws)
 	if (dws->mem_ops.exec_op)
 		master->mem_ops = &dws->mem_ops;
 	master->max_speed_hz = dws->max_freq;
-	master->flags = SPI_MASTER_GPIO_SS;
+	master->flags = SPI_CONTROLLER_GPIO_SS;
 	master->auto_runtime_pm = true;
 
 	/* Get default rx sample delay */
